### ISSUE TITLE
[Bug] dbt deps fails to install specific version of dbt-utils

### ISSUE THREAD

**Issue by ram-varkhedi** [dbt-labs/dbt-core#10305]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I have been facing issues trying to run dbt locally on my Mac. The issue started after the recent MacOS update. I am running Sonoma 14.5. I have upgraded Xcode to 15.3. My package.yml has the following dependencies. 

```yaml
packages:
  - git: https://github.com/dbt-labs/dbt-utils.git
    revision: 0.8.6

  - git: https://github.com/dbt-labs/dbt-external-tables.git
    revision: 0.8.0
```

On executing dbt deps, I get the following error. Pasting the contents from dbt.log 

```
[0m11:49:52.132100 [debug] [MainThread]: STDERR: "b''"
[0m11:49:52.132439 [debug] [MainThread]: Executing "git reset --hard origin/0.8.6"
[0m11:49:52.164114 [debug] [MainThread]: STDOUT: "b''"
[0m11:49:52.164755 [debug] [MainThread]: STDERR: "b"fatal: ambiguous argument 'origin/0.8.6': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'\n""
```

### Expected Behavior

The required packages should be installed 

### Steps To Reproduce

Execute dbt deps 

### Relevant log output

```shell
[0m11:49:47.508683 [debug] [MainThread]: Set downloads directory='/var/folders/zx/qdzkls9j34l_zk6n0c1bvpzw0000gp/T/dbt-downloads-o5zsrfya'
[0m11:49:47.515346 [debug] [MainThread]: Executing "git clone --depth 1 https://github.com/dbt-labs/dbt-utils.git b03949bb53de3439f6a8057088ba4032"
[0m11:49:49.638576 [debug] [MainThread]: STDOUT: "b''"
[0m11:49:49.639386 [debug] [MainThread]: STDERR: "b"Cloning into 'b03949bb53de3439f6a8057088ba4032'...\n""
[0m11:49:49.639857 [debug] [MainThread]: Pulling new dependency b03949bb53de3439f6a8057088ba4032.
[0m11:49:49.640123 [debug] [MainThread]: Executing "git rev-parse HEAD"
[0m11:49:49.666218 [debug] [MainThread]: STDOUT: "b'9e2b498028c03c13056e4ea58ece1368d4857458\n'"
[0m11:49:49.666920 [debug] [MainThread]: STDERR: "b''"
[0m11:49:49.667202 [debug] [MainThread]: Checking out revision 0.8.6.
[0m11:49:49.667437 [debug] [MainThread]: Executing "git remote set-branches origin 0.8.6"
[0m11:49:49.697815 [debug] [MainThread]: STDOUT: "b''"
[0m11:49:49.698418 [debug] [MainThread]: STDERR: "b''"
[0m11:49:49.698655 [debug] [MainThread]: Executing "git fetch origin --depth 1 --tags 0.8.6"
[0m11:49:52.084504 [debug] [MainThread]: STDOUT: "b''"
[0m11:49:52.085246 [debug] [MainThread]: STDERR: "b'From ssh://github.com/dbt-labs/dbt-utils\n * tag               0.8.6      -> FETCH_HEAD\n * [new tag]         0.0.1      -> 0.0.1\n * [new tag]         0.1.0      -> 0.1.0\n * [new tag]         0.1.1      -> 0.1.1\n * [new tag]         0.1.10     -> 0.1.10\n * [new tag]         0.1.11     -> 0.1.11\n * [new tag]         0.1.12     -> 0.1.12\n * [new tag]         0.1.13     -> 0.1.13\n * [new tag]         0.1.14     -> 0.1.14\n * [new tag]         0.1.15     -> 0.1.15\n * [new tag]         0.1.16     -> 0.1.16\n * [new tag]         0.1.17     -> 0.1.17\n * [new tag]         0.1.18     -> 0.1.18\n * [new tag]         0.1.19     -> 0.1.19\n * [new tag]         0.1.2      -> 0.1.2\n * [new tag]         0.1.20     -> 0.1.20\n * [new tag]         0.1.21     -> 0.1.21\n * [new tag]         0.1.22     -> 0.1.22\n * [new tag]         0.1.23     -> 0.1.23\n * [new tag]         0.1.24     -> 0.1.24\n * [new tag]         0.1.25     -> 0.1.25\n * [new tag]         0.1.3      -> 0.1.3\n * [new tag]         0.1.4      -> 0.1.4\n * [new tag]         0.1.5      -> 0.1.5\n * [new tag]         0.1.6      -> 0.1.6\n * [new tag]         0.1.7      -> 0.1.7\n * [new tag]         0.1.8      -> 0.1.8\n * [new tag]         0.1.9      -> 0.1.9\n * [new tag]         0.2.0      -> 0.2.0\n * [new tag]         0.2.1      -> 0.2.1\n * [new tag]         0.2.2      -> 0.2.2\n * [new tag]         0.2.3      -> 0.2.3\n * [new tag]         0.2.4      -> 0.2.4\n * [new tag]         0.2.5      -> 0.2.5\n * [new tag]         0.3.0      -> 0.3.0\n * [new tag]         0.4.0      -> 0.4.0\n * [new tag]         0.4.1      -> 0.4.1\n * [new tag]         0.5.0      -> 0.5.0\n * [new tag]         0.5.1      -> 0.5.1\n * [new tag]         0.6.0      -> 0.6.0\n * [new tag]         0.6.1      -> 0.6.1\n * [new tag]         0.6.2      -> 0.6.2\n * [new tag]         0.6.3      -> 0.6.3\n * [new tag]         0.6.4      -> 0.6.4\n * [new tag]         0.6.5      -> 0.6.5\n * [new tag]         0.6.6      -> 0.6.6\n * [new tag]         0.7.0      -> 0.7.0\n * [new tag]         0.7.1      -> 0.7.1\n * [new tag]         0.7.2      -> 0.7.2\n * [new tag]         0.7.3      -> 0.7.3\n * [new tag]         0.7.4      -> 0.7.4\n * [new tag]         0.7.4-b1   -> 0.7.4-b1\n * [new tag]         0.7.4b1    -> 0.7.4b1\n * [new tag]         0.7.5      -> 0.7.5\n * [new tag]         0.7.6      -> 0.7.6\n * [new tag]         0.8.0      -> 0.8.0\n * [new tag]         0.8.1      -> 0.8.1\n * [new tag]         0.8.2      -> 0.8.2\n * [new tag]         0.8.3      -> 0.8.3\n * [new tag]         0.8.4      -> 0.8.4\n * [new tag]         0.8.5      -> 0.8.5\n * [new tag]         0.8.6      -> 0.8.6\n * [new tag]         0.9.0      -> 0.9.0\n * [new tag]         0.9.1      -> 0.9.1\n * [new tag]         0.9.2      -> 0.9.2\n * [new tag]         0.9.5      -> 0.9.5\n * [new tag]         0.9.6      -> 0.9.6\n * [new tag]         1.0.0      -> 1.0.0\n * [new tag]         1.0.0-b1   -> 1.0.0-b1\n * [new tag]         1.0.0-b2   -> 1.0.0-b2\n * [new tag]         1.0.0-rc1  -> 1.0.0-rc1\n * [new tag]         1.1.0      -> 1.1.0\n * [new tag]         1.1.1      -> 1.1.1\n * [new tag]         1.2.0      -> 1.2.0\n * [new tag]         1.2.0-rc1  -> 1.2.0-rc1\n'"
[0m11:49:52.085640 [debug] [MainThread]: Executing "git tag --list"
[0m11:49:52.131290 [debug] [MainThread]: STDOUT: "b'0.0.1      0.1.18     0.1.5      0.3.0      0.6.6      0.8.1      1.0.0\n0.1.0      0.1.19     0.1.6      0.4.0      0.7.0      0.8.2      1.0.0-b1\n0.1.1      0.1.2      0.1.7      0.4.1      0.7.1      0.8.3      1.0.0-b2\n0.1.10     0.1.20     0.1.8      0.5.0      0.7.2      0.8.4      1.0.0-rc1\n0.1.11     0.1.21     0.1.9      0.5.1      0.7.3      0.8.5      1.1.0\n0.1.12     0.1.22     0.2.0      0.6.0      0.7.4      0.8.6      1.1.1\n0.1.13     0.1.23     0.2.1      0.6.1      0.7.4-b1   0.9.0      1.2.0\n0.1.14     0.1.24     0.2.2      0.6.2      0.7.4b1    0.9.1      1.2.0-rc1\n0.1.15     0.1.25     0.2.3      0.6.3      0.7.5      0.9.2\n0.1.16     0.1.3      0.2.4      0.6.4      0.7.6      0.9.5\n0.1.17     0.1.4      0.2.5      0.6.5      0.8.0      0.9.6\n'"
[0m11:49:52.132100 [debug] [MainThread]: STDERR: "b''"
[0m11:49:52.132439 [debug] [MainThread]: Executing "git reset --hard origin/0.8.6"
[0m11:49:52.164114 [debug] [MainThread]: STDOUT: "b''"
[0m11:49:52.164755 [debug] [MainThread]: STDERR: "b"fatal: ambiguous argument 'origin/0.8.6': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'\n""
[0m11:49:52.164984 [debug] [MainThread]: command return code=128
[0m11:49:52.165802 [error] [MainThread]: Encountered an error:
Internal Error
  Error checking out spec='0.8.6' for repo https://github.com/dbt-labs/dbt-utils.git
  fatal: ambiguous argument 'origin/0.8.6': unknown revision or path not in the working tree.
  Use '--' to separate paths from revisions, like this:
  'git <command> [<revision>...] -- [<file>...]'
[0m11:49:52.166519 [debug] [MainThread]: Command `dbt deps` failed at 11:49:52.166394 after 4.76 seconds
[0m11:49:52.166864 [debug] [MainThread]: Sending event: {'category': 'dbt', 'action': 'invocation', 'label': 'end', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x1056f6530>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x107a2ef80>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x107a2e590>]}
[0m11:49:52.167201 [debug] [MainThread]: Flushing usage events
```

### Environment

```markdown
- OS: MacOS
- Python: 3.10
- dbt: 1.6.15
```

### Which database adapter are you using with dbt?

bigquery

### Additional Context

NA

**Comment by ram-varkhedi** [comment]
The error seems to be this statement : "git reset --hard origin/0.8.6" . When I tried running dbt deps on my team members machine, I see the following in the logs "git reset --hard tags/0.8.6"

**Comment by dbeatty10** [comment]
Thanks for reporting this @ram-varkhedi !

Could you do a `git --version` and share the output?

**Comment by ram-varkhedi** [comment]
@dbeatty10 I am using git version 2.39.3 (Apple Git-146)

**Comment by ram-varkhedi** [comment]
I was able to resolve the error. My gitconfig had a setting the display the branches in column mode (git config --global column.ui always). I am guessing this was messing up with the output of `git tag --list ` . I will close this issue for now.

**Comment by dbeatty10** [comment]
Thanks for following up with that info @ram-varkhedi 🧠

Do you know if this setting was applied automatically during the software update? Or was it applied some other way?

If it's an automatic thing, we may way to take a look at how we can detect the setting or catch the error and give more specific information in the error message.

**Comment by ram-varkhedi** [comment]
@dbeatty10 this wasn't part of the software update. I had executed the git config command to enable this setting. I have tested my hypotheses by enabling the column listing on dbt-downloads repo (`git config --local column.ui always` ). The output of `git tag --list ` is columnar now. 

0.0.1      0.1.14     0.1.20     0.1.4      0.2.1      0.4.1      0.6.4      0.7.4      0.8.2      0.9.2      1.1.0
0.1.0      0.1.15     0.1.21     0.1.5      0.2.2      0.5.0      0.6.5      0.7.4-b1   0.8.3      0.9.5      1.1.1
0.1.1      0.1.16     0.1.22     0.1.6      0.2.3      0.5.1      0.6.6      0.7.4b1    0.8.4      0.9.6      1.2.0
0.1.10     0.1.17     0.1.23     0.1.7      0.2.4      0.6.0      0.7.0      0.7.5      0.8.5      1.0.0      1.2.0-rc1
0.1.11     0.1.18     0.1.24     0.1.8      0.2.5      0.6.1      0.7.1      0.7.6      0.8.6      1.0.0-b1
0.1.12     0.1.19     0.1.25     0.1.9      0.3.0      0.6.2      0.7.2      0.8.0      0.9.0      1.0.0-b2
0.1.13     0.1.2      0.1.3      0.2.0      0.4.0      0.6.3      0.7.3      0.8.1      0.9.1      1.0.0-rc1

The columnar display can be overridden by `git tag --no-column`.  The ouput is similar to `git tag --list `.  Having dbt deps issue a `git tag --no-column` would make it resilient if any user had enabled the columnar display setting.
